app/src/main/java/com/akshay/newsapp/koin/modules.kt

```diff
@@ -0,0 +1,44 @@
+package com.akshay.newsapp.koin
+
+import com.akshay.newsapp.db.NewsDatabase
+import com.akshay.newsapp.repo.NewsRepository
+import com.akshay.newsapp.ui.viewmodel.NewsArticleViewModel
+import org.koin.android.ext.koin.androidApplication
+import org.koin.androidx.viewmodel.dsl.viewModel
+import org.koin.dsl.module
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+val appModule = module {
+
+	// Retrofit Instance
```

Comment:
I think we should re-use the existing module classes and replace the existing dagger code to koin. It will also be a nice abstraction and also show to people how easy it to switch to Koin :)
Let me know your thoughts.

Reply:
Just to give an example of what I mean.

We will keep the `NewsServiceModule.kt` and put 
```
...
module {
    single {
         retrofit()
    }
}
...
```

Similarly for all things so they are clearly categorised

Reply:
Do you mean to say have multiple modules (e.g databaseModule, serviceModule etc) instead of one (here, appModule) ? 
I kept it in a single file just to make it a one stop file to have a look at all the instances at once.

Reply:
Yes exactly. I agree with your point of having one stop file but as the project grows the file grows with all the instances and also becames hard to modularise in the future.

Reply:
I understand. Let me see if we can easily split up the modules.

Reply:
Done with this one 👍 Marking it as resolved as of now. Will push all the changes together later.